h2o-genmodel/src/main/java/hex/genmodel/algos/glrm/GlrmMojoModel.java

```diff
@@ -70,54 +74,52 @@ public GlrmMojoModel(String[] columns, String[][] domains, String responseColumn
     return _ncolX;
   }
 
+  public double[] setAlphas(int numAlpha) {
+    double[] alphas = new double[numAlpha];
+    double alpha = 1.0;
+    for (int index=0; index < numAlpha; index++) {
+      alpha *= DOWN_FACTOR;
+      alphas[index] = alpha;
+    }
+    return alphas;
+  }
+
   public double[] score0(double[] row, double[] preds, long seedValue) {
     assert row.length == _ncolA;
     assert preds.length == _ncolX;
     assert _nrowY == _ncolX;
     assert _archetypes.length == _nrowY;
     assert _archetypes[0].length == _ncolY;
-    double alpha=1.0;  // reset back to 1 for each row
+    _allAlphas = setAlphas(_numAlphaFactors);
```

Comment:
can be done in the reader instead of for every row

Reply:
Done